Add support for IPInfusion OcNOS commit model

pklimai opened the pull request:
As discussed in issue #3590, the ipinfusion_ocnos module does not currently implement commit() method, however, modern versions of OcNOS require commits to apply configuration.

This PR implements the required commit() method. In addition, confirm_commit() method for confirm-commit operation and abort_transaction() method for abort-transaction operation are implemented. The send_config_set() method is slightly modified to raise exceptions in case config load errors are detected.

I tested this code with OcNOS 6.3, 6.4 and 6.5 versions.

pklimai commented:
Please LMK if you need anything else in regards to this PR

pklimai commented:
Hi @ktbyers, can you please approve the workflow?

pklimai commented:
Hi @ktbyers can you please approve? Do you need anything else to merge this PR?

ktbyers commented:
I will try to review this in the next week.

ktbyers commented on netmiko/ipinfusion/ipinfusion_ocnos.py:
```diff
@@ -28,15 +28,156 @@ def session_preparation(self) -> None:
+        """Send config command(s). Requires separate calling of commit to apply."""
+
+        # Do not exit config mode - it won't allow until you commit or abort transaction
+        output = super().send_config_set(*args, **kwargs, exit_config_mode=False)
```
We probably should do the following here:

```
# Default 'exit_config_mode' to False unless it is explicitly overwritten
exit_config_mode = kwargs.get("exit_config_mode", False)
```

ktbyers replied:
On error_markers, this probably should be outside of Netmiko directly and should just be called with existing `error_pattern` argument. Additionally, I am reluctant to change the default behavior for a specific driver (i.e. to default to having the error patterns specified and active in the driver).

So I would recommend we remove the error_markers and you can implement this in your Netmiko calling code as something like:

```python
        error_markers = [
            "% Invalid input detected at '^' marker.",
            "% Parameter not configured",
            "% Ambiguous command:",
        ]
       err_pattern1, err_pattern2, err_pattern3 = [re.escape(marker) for marker in error_markers]
       error_pattern(rf"(?:{err_pattern1}|{error_pattern2}|{err_pattern3})")
```

Then make your send_config_set() and pass in the `error_pattern` argument.

Assuming we do this, then `ipinfusion_ocnos` send_config_set would become just the `exit_config_mode` change and the `super()` call.

ktbyers commented on netmiko/ipinfusion/ipinfusion_ocnos.py:
```diff
@@ -28,15 +28,156 @@ def session_preparation(self) -> None:
+        default (no options):
+            command_string = commit
+        confirm and confirm_delay:
+            command_string = commit confirmed <confirm_delay>
```
We should remove `commit confirm` from the `commit` method. These should be implemented as a private method as per the communication in https://github.com/ktbyers/netmiko/issues/3590

ktbyers replied:
Same with `confirm_delay` this argument should be moved to a private `_commit_confirm()` method.

ktbyers replied:
Or worded slightly differently, Netmiko's `commit()` method only commits previous changes--it does not support `commit confirm`, this would need to be done entirely in private methods.

pklimai commented:
Thanks @ktbyers ! I made modifications as per your suggestions, please check.

ktbyers commented:
@pklimai My bad on the `commit`...I forgot that we added this `confirm` and `confirm_delay` arguments into the Juniper driver. I will revert your changes back to what you had earlier on this.

I actually remembered this while semi-sleeping last night :-)

pklimai commented:
@ktbyers Thanks for clarification, indeed I implemented commit() similar to how it was done for Junos. Now I reverted and just fixed send_config_set(). Please check.

ktbyers commented:
@pklimai Thanks for contributing this.

Note, I am going to change `description` argument to be `comment` as that is the argument I used with most of the other drivers that use commit. Let me know if you see any issues with this.

pklimai commented:
Thank you very much @ktbyers. I am fine with using `comment`.
